src/Storages/Hive/StorageHive.cpp

```diff
@@ -151,13 +151,26 @@ class StorageHiveSource : public SourceWithProgress, WithContext
         {
             if (!reader)
             {
+                if (current_file_remained_rows) [[unlikely]]
```

Comment:
why unlikely? looks like it can happen if number of rows exceeds max block size

Reply:
I think that all the cols to read are partition keys maybe rare.